frame/governance-registry/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+//! # Governance Registry Pallet
+//!
+//! Is used to add new assets into chain.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+pub mod weights;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::FullCodec;
+	use frame_support::{dispatch::DispatchResultWithPostInfo, pallet_prelude::*};
+	use frame_system::{ensure_root, pallet_prelude::OriginFor};
+	use scale_info::TypeInfo;
+
+	use crate::weights::WeightInfo;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type AssetId: Encode
+			+ Decode
+			+ Clone
+			+ core::fmt::Debug
+			+ Eq
+			+ Default
+			+ TypeInfo
+			+ FullCodec;
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	type OriginsByAssetId<T: Config> =
+		StorageMap<_, Twox64Concat, T::AssetId, StorageOrigin<T::AccountId>, ValueQuery>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NoneError,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		Set { asset: T::AssetId, value: T::AccountId },
+		GrantRoot { asset: T::AssetId },
+		Remove { asset: T::AssetId },
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Sets the value of an asset to the signed account id. Only callable by root.
+		#[pallet::weight(T::WeightInfo::set())]
+		pub fn set(
+			origin: OriginFor<T>,
+			asset: T::AssetId,
+			value: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			OriginsByAssetId::<T>::insert(asset.clone(), StorageOrigin::Signed(value.clone()));
+			Self::deposit_event(Event::<T>::Set { asset, value });
+			Ok(().into())
+		}
+
+		/// Sets the value of an asset to root. Only callable by root.
+		#[pallet::weight(T::WeightInfo::grant_root())]
+		pub fn grant_root(origin: OriginFor<T>, asset: T::AssetId) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			OriginsByAssetId::<T>::insert(asset.clone(), StorageOrigin::Root);
+			Self::deposit_event(Event::<T>::GrantRoot { asset });
+			Ok(().into())
+		}
+
+		/// Removes mapping of an asset_id. Only callable by root.
+		#[pallet::weight(T::WeightInfo::remove())]
+		pub fn remove(origin: OriginFor<T>, asset: T::AssetId) -> DispatchResultWithPostInfo {
```

Comment:
we have some build in ids (those in CurrencyId). What would it mean if I pass some id which is build in into code here?

Reply:
It removes a governance mapping, which is later used in pallet-democracy.

Reply:
ok, so we have some build in assets, with 1,2,3,4.. small numbers as const in Rust codes. Will governance be on these either?

Reply:
Pica will be mapped to Root, rest can be set, but most likely not.